pyproject.toml

```diff
@@ -21,11 +21,14 @@ lint.select = [
     "SIM2", # simplify boolean expressions
     "SIM4", # simplify if statements
     "SIM9", # simplify dict values
-    "W" # pycodestyle warnings
+    "UP006",
+    "W", # pycodestyle warnings
 ]
 lint.ignore = [
     # rules recommended to be ignored when using ruff format
-    "E501", "W191", "W291", "W293"
+    "E501", "W191",
+    # ignore the rule banning Optional[X] in favor of X | None, we are fine with Optional for now
+    "UP007",
```

Comment:
No need to ignore this, we do no not enable the whole UP. Or maybe we should enable the whole UP and not only UP006?

Reply:
I want to play with this in the follow-up. I commented UP007 because it is commented on in Quiver. I plan to check other UP rules, but first I need to check if it is required to have python-version set up to the minimum supported one.